packages/metal-dom/src/events.js

```diff
@@ -2,6 +2,54 @@
 
 import { registerCustomEvent } from './dom';
 import features from './features';
+import * as KEYMAP from './keyConstants';
+
+const REGEX_EVENT = /^(key+(down|up|press)):((,?[a-z])+)/;
+
+/**
+ * Returns the relative keyCode from each key name in an array.
+ * @param {Array<string>} keyNames An array with the key names.
+ */
+function convertKeynamesToKeyCode_(keyNames) {
+  let keys = {};
+
+  keyNames.forEach((value, index) => {
+    keys[KEYMAP[value.toUpperCase()]]= true;
+  });
+
+  return keys;
+}
+
+/**
+ * Creates an event configuration object to deail with keyboard events.
+ * @param {!Array} MatchedRegexInfo The information extracted from the regex
+ * that has matched with the parameterized event name.
+ * @param {object} The custom keyboard event configuration that has all the
+ * necessary informations for the handler function.
+ */
+function createCustomKeyboardEventConfig(MatchedRegexInfo) {
+  return {
+    event: true,
+    delegate: true,
+    keys: convertKeynamesToKeyCode_(MatchedRegexInfo[3].split(',')),
+    handler: handlerKeys,
+    originalEvent: MatchedRegexInfo[1]
+  }
+}
+
+/**
+ * Original listener wrapper. It checks if the pressed key matchs with the given
+ * keys and it trigger the original listener function.
+ * @param {!function(!Object)} callback The original listener that will called
+ * when the event is triggered and if the pressed key and the given keys alias
+ * matches.
+ * @param {event} Event object.
+ */
+function handlerKeys(callback, event) {
+  if (this.keys[event.keyCode]) {
```

Comment:
We always set the `customType` property of the event before triggering it, in case the developer wants to know  it. It'd be good to do this here as well, same way we do for the other custom events on this file: `event.customType = eventName`. Note that `eventName` should be the name that was passed to `dom.on` or `dom.delegate`, for example: `keydown:enter,esc`.